cmd/common/registry.go

```diff
@@ -425,6 +456,15 @@ func addFlags(cmd *cobra.Command, params *params.Params) {
 	for _, p := range *params {
 		desc := p.Description
 
+		// Special case: if a DefaultValue starts with an exclamation mark, we
+		// look up the default value in the runtime
```

Comment:
Yeah, I'm also not 100% convinced about this. Right now we're using it for the default kubernetes namespace.
I think a better solution would be to have another field in the Params that gives context (not just a name). For example, we could have a field like `ContentHint` with a value of `k8s:namespace`. The registry can read that and just fill it out. This would also help a great deal with other frontends or even our cobra frontend improved (think autocompletion for params).

Reply:
Having it in a different (dedicated) field would make it clearer.